dgraph/cmd/zero/raft.go

```diff
@@ -88,11 +88,11 @@ func (n *node) AmLeader() bool {
 // {2 bytes Node ID} {4 bytes for random} {2 bytes zero}
 func (n *node) initProposalKey(id uint64) error {
 	x.AssertTrue(id != 0)
-	b := make([]byte, 8)
-	if _, err := rand.Read(b); err != nil {
+	random4Bytes := make([]byte, 4)
```

Comment:
let's keep the 17th bit from the left as 0 when initializing the proposal key. That will ensure that the node ID bytes won't be modified during calls to uniqueKey.